### PR TITLE
Update Thing Client to fetch pending things

### PR DESCRIPTION
**Pull request by miburf**
Update Thing Client to fetch pending things

### CODE REVIEW

File: MHVLib/Clients/MHVThingClient.m [hunk 1]
```diff
@@ -162,13 +182,48 @@ - (void)getThingsWithQueries:(MHVThingQueryCollection *)queries
          else
          {
              MHVThingQueryResults *queryResults = [self thingQueryResultsFromResponse:response];
```
Comment (namalu) [inline MHVLib/Clients/MHVThingClient.m:184]: I think that we need to check if the queryResults parsed correctly by verifying that it is not nil. If it is nil we should return an error so the caller knows that there is a problem with the response.

File: MHVLib/Clients/MHVThingClient.m [hunk 2]
```diff
@@ -162,13 +182,48 @@ - (void)getThingsWithQueries:(MHVThingQueryCollection *)queries
...
              else
              {
-                 completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetThings results could not be extracted."]);
+                 if (results)
+                 {
+                     completion(results, nil);
+                 }
+                 else
+                 {
+                     completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetThings results could not be extracted."]);
```
Comment (namalu) [inline MHVLib/Clients/MHVThingClient.m]: I'm not quite sure what this error description means. We should be as detailed as possible with our error descriptions so debugging is easier for us and consumers of the SDK.

File: MHVLib/Types/MHVThingQueryResult.m [hunk 3]
```diff
@@ -199,4 +199,33 @@ - (instancetype)init
...
+}
+
+- (void)mergeThingQueryResultCollection:(MHVThingQueryResultCollection *)collection
+{
+    for (MHVThingQueryResult *result in collection)
+    {
+        MHVThingQueryResult *existingResult = [self resultWithName:result.name];
+        if (existingResult)
+        {
+            [existingResult.things addObjectsFromArray:result.things.toArray];
```
Comment (namalu) [inline MHVLib/Types/MHVThingQueryResult.m]: We should use addObjectsFromCollection:

**Review by namalu — APPROVED** [review]
Just a few comments.

File: MHVLib/Clients/MHVThingClient.m (hunk 1 above)
Comment (miburf) [reply MHVLib/Clients/MHVThingClient.m:184]: Added.

File: MHVLib/Clients/MHVThingClient.m (hunk 2 above)
Comment (miburf) [reply MHVLib/Clients/MHVThingClient.m]: The other new one covers it, so removed; since could be no results

**Pull request merged**